add blog post about using goal templates (#291)
actualbudget/docs: blog/2023-12-20-automate.md

---
title: Automate Your Budget With Goal Templates
description: An example of how you can automate all of your budgeting by using Actual's goal template feature

> slug: 2024-01-automateyourbudget

Reviewer: <!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
slug: 2024-01-automate-your-budget-with-goal-templates
```

tags: [budgeting, goals]
authors: youngcw
hide_table_of_contents: false
---

Hello budgeters!
I hope you are all enjoying Actual as much as I do!

> As a new year approaches so to do new goals.

Reviewer: ```suggestion
As a new year is approaching so is the goal-setting and reflection season.
```

> I'm hoping to maybe help some of you out with potential new years resolutions by making budgeting as easy as possible while still reaching for your goals.

Reviewer: ```suggestion
I'm hoping to help some of you with potential new years resolutions by making budgeting as easy as possible while still reaching for your goals.
```

I believe that Actual is a great tool for this whether you have a simple budget or a complex one.

> One hurdle that many have with budgeting is that it takes constant work to stay up to date.

Reviewer: ```suggestion
One hurdle that many have with budgeting is that it takes constant effort to stay up to date.
```

Entering transactions, assigning categories, reconciling with your bank and others can hold us back.

> Some of those are just the nature of the beast, but some can streamlined.

Reviewer: <!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏not sure I understand this sentence.

> I hope to give you some pointers and strategies today to reduce some of the stress of budgeting by streamlining the budgeting processes by using goal templates.

Reviewer: ```suggestion
I hope to give you some pointers and strategies today to reduce some of the stress of budgeting by streamlining the budgeting processes with goal templates.
```

Or you could drop the remaining funds into your savings category to save even more!

Another strategy that you could employ with leftover funds is to circle back to a previously budgeted category and add more to it.
For example I have a sinking fund that I would like to budget more to if I have leftover funds available after my base fun money is budgeted.
To do this I have two template lines at different priorities it looks like:
* `#template-30 100 up to 1500`
* `#template-60 100 up to 1500`
This adds 100 to this category on the first pass and will happen before my fun categories.
Then gets a second chance to add more after my fun categories by having a higher priority level.

## Conclusion

With this strategy for goal templates, starting a new month budget takes just a few seconds.
I hope that there was something in here that was useful to you in your budgeting journey.

If budgeting feels daunting, don't worry.
It is hard to get going and can be hard if things are tight, but taking the time to get on top of things now will give you more financial freedom in the future.

I wish you a merry holiday season and a happy new year!

Happy budgeting

> -- youngcw

Reviewer: <!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I think you can remove this since your name is already signed at the top of the post.

---
title: Automate Your Budget With Goal Templates
description: An example of how you can automate all of your budgeting by using Actual's goal template feature

> slug: 2023-12-automate-your-budget-with-goal-templates

Reviewer: FYI: might want to update this date (and filename) before merging. With the latest date.

You can do this by using `up to` like: `#template up to $100`.

#### Schedule
If you have taken the time to curate a list of schedules of expected transactions, you can automate budgeting them with the schedule template.

An example would look like `#template schedule Rent`.

This will automatically budget the money you need to pay your rent based on your `Rent` schedule.

#### Percentage
Say you want to budget a percent of your income, or a percent of what you have available.
This is useful for savings, or to keep fun spending within a limit.
Some examples of this are:

* `#template 10% of all income` to budget 10% of all income categories.
* `#template 10% of Paycheck` to only include the "Paycheck" category in the calculation.

### Template Priorities
Okay, so you understand the templates pretty well now.
Now for, in my opinion, the most useful part of templates.
We can set priorities!
Setting a priority to the templates does two things for us
1. It tells Actual to not budget more money then shows in your `To Budget`.  This is not the case by default.

> 2. I tells Actual what order to apply the templates.

Reviewer 2: It tells Actual in what (or in which) order to apply the templates.
